repositories/org_repository.go

```diff
@@ -0,0 +1,142 @@
+package repositories
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+const (
+	OrgNameLabel   = "cloudfoundry.org/org-name"
+	SpaceNameLabel = "cloudfoundry.org/space-name"
+)
+
+type OrgRecord struct {
+	Name      string
+	GUID      string
+	Suspended bool
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type SpaceRecord struct {
+	Name             string
+	GUID             string
+	OrganizationGUID string
+	CreatedAt        time.Time
+	UpdatedAt        time.Time
+}
+
+type OrgRepo struct {
+	rootNamespace    string
+	privilegedClient client.Client
+}
+
+func NewOrgRepo(rootNamespace string, privilegedClient client.Client) *OrgRepo {
+	return &OrgRepo{
+		rootNamespace:    rootNamespace,
+		privilegedClient: privilegedClient,
+	}
+}
+
+func (r *OrgRepo) FetchOrgs(ctx context.Context, names []string) ([]OrgRecord, error) {
+	subnamespaceAnchorList := &v1alpha2.SubnamespaceAnchorList{}
+
+	options := []client.ListOption{client.InNamespace(r.rootNamespace)}
+	if len(names) > 0 {
+		namesRequirement, err := labels.NewRequirement(OrgNameLabel, selection.In, names)
+		if err != nil {
+			return nil, err
+		}
+		namesSelector := client.MatchingLabelsSelector{
+			Selector: labels.NewSelector().Add(*namesRequirement),
+		}
+		options = append(options, namesSelector)
+	}
+
+	err := r.privilegedClient.List(ctx, subnamespaceAnchorList, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	records := []OrgRecord{}
+	for _, anchor := range subnamespaceAnchorList.Items {
+		records = append(records, OrgRecord{
+			Name:      anchor.Labels[OrgNameLabel],
+			GUID:      string(anchor.UID),
```

Comment:
Want to align my understanding here. 
Org names or space name are captured as `labels` on these records and NOT as `metadata.name`.  

In the acceptance steps @kieron-dev has described what does `org1-xxx` or `space1-xxx` correspond to? A GUID?

Reply:
Org names are only unique within a foundation root. We reserve the possibility of having multiple foundations on the same cluster, although that's not something we're considering at the moment. And space names are only unique within an org. As orgs and spaces both translate to namespaces which need cluster-wide unique names, and the namespace names match the subnamespaceanchor names, we have to make the name unique somehow. So xxx and yyy indicates something to make the name unique, and we store the requested name, i.e. the org our space name, in a label.

Reply:
Thanks @kieron-dev for clarification. This aligns with my understanding.

Reply:
Added comments to the [Create Orgs PR](https://github.com/cloudfoundry/cf-k8s-api/pull/119#pullrequestreview-770906888) around generating unique names that might affect change to this code.